analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/HashesShouldHaveUnpredictableSaltTest.cs

```diff
@@ -20,70 +20,130 @@
-                .Verify();
+    [TestMethod]
+    public void HashesShouldHaveUnpredictableSalt_Sonar_CSharp11() =>
+        sonar
+            .AddPaths("HashesShouldHaveUnpredictableSalt.CSharp11.cs")
+            .WithOptions(ParseOptionsHelper.FromCSharp11)
+            .Verify();
 
-        [TestMethod]
-        public void HashesShouldHaveUnpredictableSalt_CSharp11() =>
-            sonar.AddPaths("HashesShouldHaveUnpredictableSalt.CSharp11.cs")
-                .WithOptions(ParseOptionsHelper.FromCSharp11)
-                .Verify();
+    [Ignore] // TODO: remove after S2053 is implemented with the new SE engine
+    [TestMethod]
+    public void HashesShouldHaveUnpredictableSalt_Roslyn_CSharp11() =>
+        roslynCS
+            .AddPaths("HashesShouldHaveUnpredictableSalt.CSharp11.cs")
+            .WithOptions(ParseOptionsHelper.FromCSharp11)
+            .Verify();
 
 #endif
 
-        [TestMethod]
-        public void HashesShouldHaveUnpredictableSalt_LocationContext_Equals()
-        {
-            var tree = SyntaxFactory.ParseSyntaxTree("public class Test {}");
-            var location = Location.Create(tree, TextSpan.FromBounds(0, 6));
-
-            var context1 = new HashesShouldHaveUnpredictableSalt.LocationContext(location, "message");
-            var context2 = new HashesShouldHaveUnpredictableSalt.LocationContext(location, "2nd message");
-            var context3 = new HashesShouldHaveUnpredictableSalt.LocationContext(null, "message");
-            var context4 = new HashesShouldHaveUnpredictableSalt.LocationContext(location, "message");
-
-            context1.Equals(null).Should().BeFalse();
-            context1.Equals(new object()).Should().BeFalse();
-            context1.Equals(context2).Should().BeFalse();
-            context1.Equals(context3).Should().BeFalse();
-            context1.Equals(context1).Should().BeTrue();
-            context1.Equals(context4).Should().BeTrue();
-        }
+    [TestMethod]
+    public void HashesShouldHaveUnpredictableSalt_Sonar_LocationContext_Equals()
+    {
+        var tree = SyntaxFactory.ParseSyntaxTree("public class Test {}");
+        var location = Location.Create(tree, TextSpan.FromBounds(0, 6));
+
+        var context1 = new HashesShouldHaveUnpredictableSalt.LocationContext(location, "message");
+        var context2 = new HashesShouldHaveUnpredictableSalt.LocationContext(location, "2nd message");
+        var context3 = new HashesShouldHaveUnpredictableSalt.LocationContext(null, "message");
+        var context4 = new HashesShouldHaveUnpredictableSalt.LocationContext(location, "message");
+
+        context1.Equals(null).Should().BeFalse();
+        context1.Equals(new object()).Should().BeFalse();
+        context1.Equals(context2).Should().BeFalse();
+        context1.Equals(context3).Should().BeFalse();
+        context1.Equals(context1).Should().BeTrue();
+        context1.Equals(context4).Should().BeTrue();
```

Comment:
I don't understand fully the purpose of this test. :/ 
Why is is needed and is it also needed for the migrated rule?

Reply:
`LocationContext` is a nested class inside the Sonar version of `HashesShouldHaveUnpredictableSalt`. This UT just validates the implemented `Equals` method inside it, to test that it works correctly and to have code coverage.
I don't plan to use such a type in the Roslyn implementation for now, so I'm not making a copy of this for the new rule implementation.

Reply:
great thanks!